pkg/cvo/metrics.go

```diff
@@ -196,12 +196,22 @@ func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
 				ch <- g
 			}
 		}
-		if failing != nil && failing.Status == configv1.ConditionTrue {
-			g := m.version.WithLabelValues("failure", current.Version, current.Image, completed.Version)
-			if failing.LastTransitionTime.IsZero() {
-				g.Set(0)
+		if failing != nil {
+			if failing.Status == configv1.ConditionTrue {
+				g := m.version.WithLabelValues("failure", current.Version, current.Image, completed.Version)
+				if failing.LastTransitionTime.IsZero() {
+					g.Set(0)
+				} else {
+					g.Set(float64(failing.LastTransitionTime.Unix()))
+				}
+				ch <- g
+			}
+
+			g := m.clusterOperatorConditions.WithLabelValues("cluster-version", "Failing", string(failing.Reason))
```

Comment:
Also we should probably just push all the conditions from the clusterversion object like https://github.com/openshift/cluster-version-operator/blob/b69cd4847fab21a74a15544cc66a1d69dab63f31/pkg/cvo/metrics.go#L268-L279. The conditions are same as other clusteroperators.. probably here https://github.com/openshift/cluster-version-operator/blob/b69cd4847fab21a74a15544cc66a1d69dab63f31/pkg/cvo/metrics.go#L281

Reply:
> Also we should probably just push all the conditions...

Done with d881086 -> 240244c.